frontend/app/components/redesign/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect, useCallback } from 'react'
+
+export const useCopyToClipboard = (link: string) => {
```

Comment:
This function can be generic enough:
```suggestion
export const useCopyToClipboard = (text: string) => {
```

and pass it full `<link...>` as input.